Cargo.toml

```diff
@@ -29,86 +29,170 @@ license = "BSD-3-Clause-Clear"
-mockall = "=0.13.1"
-opentelemetry = "=0.29.1"
-paste = "=1.0"
-prost = "=0.13.5"
-prost-build = "=0.13.5"
-prost-types = "=0.13.5"
-rand = "=0.8.5"
-rayon = "=1.11.0"
-rcgen = { version = "=0.14.0", default-features = false, features = [
-    "aws_lc_rs",
-    "crypto",
-    "pem",
-    "x509-parser",
-] }
-reqwest = { version = "=0.12.22", default-features = false, features = [
-    "json",
-    "rustls-tls",
-] }
-schemars = "=0.8.22"
-serde = { version = "1.0.223", features = ["derive", "rc"] }
-serde_json = "=1.0.145"
-serial_test = "=3.2.0"
-strum = "=0.27.1"
-strum_macros = "=0.27.1"
-tempfile = "=3.20.0"
-thiserror = "=2.0.12"
-tonic-health = "=0.13.1"
-tonic-tls = "=0.3.0"
-tower = "=0.5.2"
-tower-http = "=0.6.6"
-tfhe = "=1.3.3"
-tfhe-csprng = "=0.6.0"
-tfhe-versionable = "=0.6.1"
-tfhe-zk-pok = "=0.7.2"
-tokio = { version = "=1.46.1", features = ["full"] }
-tokio-rustls = { version = "=0.26.2", default-features = false, features = [
-    "aws_lc_rs",
-] }
-tonic = "=0.13.1"
-tonic-build = "=0.13.1"
-tracing = { version = "=0.1.41", features = ["log"] }
-tracing-appender = "=0.2.3"
-tracing-opentelemetry = "=0.30.0"
-tracing-subscriber = { version = "=0.3.20", features = ["fmt", "std"] }
-trait-variant = "0.1.2"
-validator = { version = "=0.20.0", features = ["derive"] }
-zeroize = { version = "=1.8.1", features = ["zeroize_derive"] }
-x509-parser = { version = "=0.17.0", features = ["verify"] }
+# ⚠️ WORKSPACE DEPENDENCY MANAGEMENT BEST PRACTICES ⚠️
+# 
+# 1. ADD ALL NEW DEPENDENCIES HERE in the workspace root, NOT in individual member crates
+# 2. Member crates should only reference workspace dependencies using .workspace = true
+# 3. This centralized approach ensures:
+#    - Consistent versions across all crates
+#    - Easier security audits and updates
+#    - Prevention of version conflicts
+#    - Single source of truth for dependency management
+#
+# SECURITY EVALUATION CHECKLIST for new dependencies:
+# 1. Is the owner suspicious? (limited to one or few people/small companies in risky jurisdictions)
+# 2. Is the crate not particularly popular? (check download counts on crates.io)
+# 3. Is there an unusual jump in package versions? (e.g., 0.1.0 -> 3.0.0)
+# 4. Is documentation lacking? (check docs.rs and GitHub repo)
+# 5. For crypto/security-critical deps: Has it been audited? Is it from a reputable org?
+# 
+# Risk levels:
+# - LOW RISK: Well-known maintainers, high popularity (>1M downloads), good docs
+# - MEDIUM RISK: Individual maintainers, low popularity, security-critical, or needs audit
+# - HIGH RISK: Deprecated, suspicious ownership, undocumented, or major version jumps
+#
+# MAINTAIN ALPHABETICAL ORDER for all dependencies within each section
 
-alloy-dyn-abi = "=1.3.1"
-alloy-primitives = "=1.3.1"
-alloy-sol-types = "=1.3.1"
-alloy-signer = "=1.0.30"
-alloy-signer-local = "=1.0.30"
+# Internal Zama modules
+backward-compatibility = { path = "./backward-compatibility" }  # Internal backward compatibility testing
```

Comment:
If we are also assigning risk to Zama dependencies, should this one not also have a `LOW RISK` marker?

Reply:
See comment in https://github.com/zama-ai/kms/pull/132#discussion_r2359732670